src/RMPI/helpers.jl

```diff
@@ -67,6 +67,7 @@ mpi_barrier(comm = mpi_comm()) = MPI.Barrier(comm)
 Compute the rank where the `key` belongs.
 """
 targetrank(key, np) = hash(key, hash(1)) % np
```

Comment:
Awesome work to find this bug! Would it make sense to restrict the type here, in order to catch or avoid such bugs in the future? Something like:
```suggestion
targetrank(key::Union{<:AbstractFockAddress,<:Integer}, np) = hash(key, hash(1)) % np
```